### PR TITLE
feat: data test id for env-override & app details

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Data test id for 
1. config map override
2. App Details
3. Secrets override

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 1]
```diff
@@ -262,7 +262,7 @@ function NodeComponent({ handleFocusTabs, externalLinks, monitoringTools, isDevt
                             )}
                         </div>
                     )}
-                    <div className="node-row m-0 resource-row">
+                    <div className="node-row m-0 resource-row" data-testid="">
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:265]: Remove the empty attribute if not in use

File: src/components/secrets/Secret.tsx [hunk 2]
```diff
@@ -779,6 +780,7 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
                     className={`form__input`}
                     placeholder={`random-secret`}
                     disabled={props.isUpdate}
+                    ata-testid="create-secrets-name-input"
```
Comment (sohel-rp) [inline src/components/secrets/Secret.tsx:783]: fix this attr

File: src/components/common/Select/Select.tsx [hunk 3]
```diff
@@ -106,9 +107,9 @@ const OptionGroup: React.SFC<OptionGroupProps> = function ({ children, label, ro
     </div>
 }
 
-function Button({ children, style = {}, rootClassName = "", arrowAsset = "" }) {
+function Button({ children, style = {}, rootClassName = "", arrowAsset = "", dataTestId="" }) {
     const { popupOpen, loading } = useSelectContext()
-    return <div className={`select-button flex ${rootClassName} ${popupOpen ? 'focused' : ''}`} style={{ ...style }}>
+    return <div className={`select-button flex ${rootClassName} ${popupOpen ? 'focused' : ''}`} style={{ ...style }} dataTestId={dataTestId}>
```
Comment (sohel-rp) [inline src/components/common/Select/Select.tsx:112]: replace dataTestId with html attr

File: src/components/common/Select/Select.tsx [hunk 4]
```diff
@@ -70,12 +71,12 @@ const Select: React.FC<SelectProps> & SelectComposition = function ({
...
     return <SelectContext.Provider value={{ selected, select, popupOpen, valueComparator, searchString, search, handleClick, disabled, loading, setLoading }}>
-        <PopupMenu onToggleCallback={isOpen => togglePopup(isOpen)} autoClose={popupOpen}>
-            <PopupMenu.Button isKebab={isKebab} disabled={disabled} tabIndex={tabIndex} rootClassName={rootClassName}>
+        <PopupMenu onToggleCallback={isOpen => togglePopup(isOpen)} autoClose={popupOpen} dataTestId={dataTestId}>
+            <PopupMenu.Button isKebab={isKebab} disabled={disabled} tabIndex={tabIndex} rootClassName={rootClassName} dataTestId={dataTestId}>
                 {button}
             </PopupMenu.Button>
             {popupOpen && AsyncComponent}
-            <PopupMenu.Body rootClassName={`select-popup ${rootClassName || ""}`} autoWidth={autoWidth}>
+            <PopupMenu.Body rootClassName={`select-popup ${rootClassName || ""}`} autoWidth={autoWidth} rootClassName="" bodyDataTestId="">
```
Comment (sohel-rp) [inline src/components/common/Select/Select.tsx:79]: Is empty expected? If not then remove.

Comment (sohel-rp) [reply src/components/common/Select/Select.tsx:79]: also rootClassName attr is duplicate

File: src/components/app/details/appDetails/SourceInfo.tsx [hunk 5]
```diff
@@ -96,11 +96,11 @@ export function SourceInfo({
                     </Tippy>
                 )}
                 {appDetails?.deploymentAppDeleteRequest && (
-                    <>
+                    <div data-testid="deleteing-argocd-pipeline">
```
Comment (sohel-rp) [inline src/components/app/details/appDetails/SourceInfo.tsx:99]: Also, check & confirm that there's no UI impact of adding this div

File: src/components/EnvironmentOverride/ConfigMapOverrides.tsx [hunk 6]
```diff
@@ -604,8 +604,8 @@ const OverrideConfigMapForm: React.FC<ConfigMapProps> = memo(function OverrideCo
                                 disabled={false}
                                 onChange={changeEditorMode}
                             >
-                                <RadioGroup.Radio value="gui">GUI</RadioGroup.Radio>
-                                <RadioGroup.Radio value="yaml">YAML</RadioGroup.Radio>
+                                <RadioGroup.Radio value="gui" dataTestId="GUI-from-config-map">GUI</RadioGroup.Radio>
+                                <RadioGroup.Radio value="yaml" dataTestId="YAML-from-config-map">YAML</RadioGroup.Radio>
```
Comment (sohel-rp) [inline src/components/EnvironmentOverride/ConfigMapOverrides.tsx:608]: use test id in small case

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx (hunk 1 above)
Comment (shivani170) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:265]: Done

File: src/components/secrets/Secret.tsx (hunk 2 above)
Comment (shivani170) [reply src/components/secrets/Secret.tsx:783]: Done

File: src/components/app/details/appDetails/SourceInfo.tsx (hunk 5 above)
Comment (shivani170) [reply src/components/app/details/appDetails/SourceInfo.tsx:99]: No impact

File: src/components/EnvironmentOverride/ConfigMapOverrides.tsx (hunk 6 above)
Comment (shivani170) [reply src/components/EnvironmentOverride/ConfigMapOverrides.tsx:608]: done

File: src/components/common/Select/Select.tsx [hunk 7]
```diff
@@ -70,8 +71,8 @@ const Select: React.FC<SelectProps> & SelectComposition = function ({
     }, { button: null, body: [], optionLength: 0, AsyncComponent: null })
 
     return <SelectContext.Provider value={{ selected, select, popupOpen, valueComparator, searchString, search, handleClick, disabled, loading, setLoading }}>
-        <PopupMenu onToggleCallback={isOpen => togglePopup(isOpen)} autoClose={popupOpen}>
-            <PopupMenu.Button isKebab={isKebab} disabled={disabled} tabIndex={tabIndex} rootClassName={rootClassName}>
+        <PopupMenu onToggleCallback={isOpen => togglePopup(isOpen)} autoClose={popupOpen} dataTestId={dataTestId}>
+            <PopupMenu.Button isKebab={isKebab} disabled={disabled} tabIndex={tabIndex} rootClassName={rootClassName} dataTestId={dataTestId}>
```
Comment (sohel-rp) [inline src/components/common/Select/Select.tsx:75]: why are we using same ids for both child & parent?

File: src/components/secrets/Secret.tsx [hunk 8]
```diff
@@ -729,6 +729,7 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
                 <div className="form-row__select-external-type flex">
                     <ReactSelect
                         placeholder="Select Secret Type"
+                        data-testid="secrets-data-type-select-dropdown"
```
Comment (sohel-rp) [inline src/components/secrets/Secret.tsx:732]: Remove as it won't have any affect. Use classNamePrefix instead.

**Review by sohel-rp — APPROVED** [review]

File: src/components/secrets/Secret.tsx (hunk 8 above)
Comment (shivani170) [reply src/components/secrets/Secret.tsx]: Done

**Review by kushagra-devtron — APPROVED** [review]

**Pull request merged**